src/Reader/Reader.php

```diff
@@ -117,23 +118,30 @@ public static function setCache(CacheStorage $cache)
      *
      * Sets the HTTP client object to use for retrieving the feeds.
      *
-     * @param  ZendHttp\Client $httpClient
+     * @param  ZendHttp\Client | Http\ClientInterface $httpClient
      * @return void
      */
-    public static function setHttpClient(ZendHttp\Client $httpClient)
+    public static function setHttpClient($httpClient)
     {
+        if ($httpClient instanceof ZendHttp\Client) {
+            $httpClient = new Http\ZendHttpClientDecorator($httpClient);
+        }
+
+        if (! $httpClient instanceof Http\ClientInterface) {
```

Comment:
`HeaderAwareClientInterface` *also* defines `get()`, but does so in order to add the second optional argument of `array $headers = []`. This is compatible with `ClientInterface::get()` (additional optional arguments do not violate contracts), but ensures that the client implementation is aware they could be passed, and theoretically will do something with them. Internally, we can *always* pass the second argument without issue; client implementations will then determine what to do with them.